apps/condo/domains/organization/integrations/sbbol/sync/requestTransactions.js

```diff
@@ -0,0 +1,184 @@
+            if (receivedTransactions) {
+                receivedTransactions.map( transaction => transactions.push(transaction))
+            } else {
+                logger.error(`Unsuccessful response to transaction request by state: ${{ 
+                    bankAccount: bankAccount.number, 
+                    statementDate,
+                    page,
+                }}`)
+            }
+
+            page++
+
+            // Checking that the response contains a link to the next page, if it is not there, then all transactions have been received
+            if (isEmpty(get(response, 'data._links', []).filter(link => link.rel === 'next'))) {
+                doNextRequest = false
+            }
+
+            // WORKFLOW_FAULT means invalid request parameters, that can occur in cases:
+            // when report is requested for date in future
+            // when report page does not exist, for example number is out of range of available pages
+            if (get(transactions, 'error.cause') === 'WORKFLOW_FAULT') doNextRequest = false
+        } while ( doNextRequest )
+
+        for (const transaction of transactions) {
+            // If SBBOL returned a transaction with an unsupported currency, do not process
+            if (!isEmpty(ISO_CODES.map( currencyName => get(transaction, 'amount.currencyName') === currencyName))) {
+                const formatedOperationDate = dayjs(transaction.operationDate).format('YYYY-MM-DD')
+                const whereConditions = {
+                    number: transaction.number,
+                    date:  formatedOperationDate,
+                    amount: transaction.amount.amount,
+                    currencyCode: transaction.amount.currencyName,
+                    purpose: transaction.paymentPurpose,
+                    isOutcome: transaction.direction === 'CREDIT',
+                    importId: transaction.uuid,
+                    importRemoteSystem: SBBOL_IMPORT_NAME,
+                }
+
+                const foundTransaction = await BankTransaction.getOne(context, {
+                    organization: { id: organizationId },
+                    account: { id: bankAccount.id },
+                    ...whereConditions,
+                })
+
+                if (!foundTransaction) {
+                    const bankIntegrationContextId = get(bankAccount, 'integrationContext.id')
+                    const createdTransaction = await BankTransaction.create(context, {
+                        organization: { connect: { id: organizationId } },
+                        account: { connect: { id: bankAccount.id } },
+                        integrationContext: { connect: { id: bankIntegrationContextId } },
+                        meta: { sbbol: transaction },
+                        ...whereConditions,
+                        ...dvSenderFields,
+                    })
+                    logger.info(`BankTransaction instance created with id: ${createdTransaction.id}`)
```

Comment:
We have an agreement to pass message to `msg` field to have it correctly wrapped by ELK.
ELK behaves in a way that it applies this object over higher-order log entry.
We should recap it with infrastructure team, btw.

```suggestion
                    logger.info({ msg: `BankTransaction instance created with id: ${createdTransaction.id}` })
```